vortex-duckdb/src/exporter/list.rs

```diff
@@ -3,86 +3,99 @@
         }
 
-        let offsets = &self.offsets.as_slice::<T>()[offset..][..(len + 1)];
-        let start_offset = offsets[0]
-            .to_u64()
-            .ok_or_else(|| vortex_err!("list offsets must fit in u64"))?;
-        let mut sum_of_list_lens = 0_u64;
+        let offsets = &self.offsets.as_slice::<O>()[offset..offset + len];
+        let sizes = &self.sizes.as_slice::<S>()[offset..offset + len];
+        debug_assert_eq!(offsets.len(), len);
+        debug_assert_eq!(sizes.len(), len);
 
-        let offsets_slice: &mut [cpp::duckdb_list_entry] =
+        // SAFETY: TODO(connor): Pretty sure that `export` needs to be `unsafe`.
+        let duckdb_list_views: &mut [cpp::duckdb_list_entry] =
             unsafe { vector.as_slice_mut::<cpp::duckdb_list_entry>(len) };
+        debug_assert_eq!(duckdb_list_views.len(), len);
 
-        for (window, destination) in offsets.windows(2).zip_eq(offsets_slice) {
-            let start = window[0]
+        for i in 0..len {
+            let offset = offsets[i]
                 .to_u64()
-                .ok_or_else(|| vortex_err!("list offsets must fit in u64"))?;
-            let end = window[1]
+                .ok_or_else(|| vortex_err!("somehow unable to convert an offset to u64"))?;
+            let length = sizes[i]
                 .to_u64()
-                .ok_or_else(|| vortex_err!("list offsets must fit in u64"))?;
-            let len = end - start;
-            sum_of_list_lens += len;
-            *destination = cpp::duckdb_list_entry {
-                offset: start - start_offset,
-                length: len,
-            };
+                .ok_or_else(|| vortex_err!("somehow unable to convert an offset to u64"))?;
+
+            debug_assert!(offset + length <= self.num_elements as u64);
+
+            duckdb_list_views[i] = cpp::duckdb_list_entry { offset, length };
         }
 
-        // TODO(DK): This calls `list_vector_reserve` once for each call to `export`. Moreover, we
-        // copy slices of the elements once for each call to `export`. Would `export` be faster if
-        // we copied all the elements on the first call to `export` so that subsequent calls need
-        // only copy the correct slice of offsets?
-        vector.list_vector_reserve(sum_of_list_lens)?;
+        // TODO(connor): This is quite bad: we have to export the entire elements array every single
+        // time export even a small slice of the `ListViewArray`.
+        vector.list_vector_reserve(self.num_elements as u64)?;
+        vector.list_vector_set_size(self.num_elements as u64)?;
+
         let mut elements_vector = vector.list_vector_get_child();
-        vector.list_vector_set_size(sum_of_list_lens)?;
-        self.elements_exporter.export(
-            usize::try_from(start_offset)?,
-            usize::try_from(sum_of_list_lens)?,
-            &mut elements_vector,
-        )
+
+        self.elements_exporter
+            .export(0, self.num_elements, &mut elements_vector)
```

Comment:
Do this after writing the elements to the child?
```suggestion
let mut elements_vector = vector.list_vector_get_child();

        self.elements_exporter
            .export(0, self.num_elements, &mut elements_vector)
            
        vector.list_vector_set_size(self.num_elements as u64)?;
```